src/validation.h

```diff
@@ -667,7 +667,22 @@ class CChainState {
     //! if we pruned.
     void PruneAndFlush();
 
-    bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams, std::shared_ptr<const CBlock> pblock) LOCKS_EXCLUDED(cs_main);
+    /**
+     * Make the best chain active, in multiple steps. The result is either failure
+     * or an activated best chain. pblock is either nullptr or a pointer to a block
+     * that is already loaded (to avoid loading it again from disk).
+     *
+     * ActivateBestChain is split into steps (see ActivateBestChainStep) so that
+     * we avoid holding cs_main for an extended period of time; the length of this
+     * call may be quite long during reindexing or a substantial reorg.
+     *
+     * May not be called with cs_main held. May not be called in a
+     * validationinterface callback.
+     */
```

Comment:
295ab2cc02f3ed5a1dc7ffc5c3f0e2d6015dd483

Could also document return value.

Reply:
This is taken care of in https://github.com/bitcoin/bitcoin/pull/16757 so I won't duplicate effort here.